### ISSUE TITLE
SAX style: Error offset should be at offending location, not after

### ISSUE THREAD

**Issue by kkaefer** [miloyip/rapidjson#363]
Currently, the location reported by `GetErrorOffset()` reports the offset *after* the symbol that failed, but it should report the location before the symbol instead. Pointing to after the symbol may lead to confusing error messages because the user may not immediately recognize that they have to look *before* the reported location, not at the reported location.

**Comment by miloyip** [comment]
I think this should need to be more specific.
I just try to find some examples:

~~~cpp
    void ParseObject(InputStream& is, Handler& handler) {
        // ...
        for (SizeType memberCount = 0;;) {
            if (is.Peek() != '"')
                RAPIDJSON_PARSE_ERROR(kParseErrorObjectMissName, is.Tell());
    }
~~~
In this error, the offset is at the character which presumed to be a '"' character but it is not. I think this is a correct offset.

There may be some errors that calculate the offset incorrectly. If you find that, you can give some examples so we can track the issue.

**Comment by kkaefer** [comment]
The internal parsing reports the correct locations, but as soon as a handler function is called that returns `false`, the position *after* the symbol is reported. This is because the symbol has to be parsed first before being handed over to the handler.

**Comment by miloyip** [comment]
Thank you for clarification. I got your point now. I can fix it by memorising the offset before calling the handler. I think it shall not affect the performance if the handler always returns `true`.

**Comment by kkaefer** [comment]
Yeah, I think so too, it'll get optimized away by the compiler in all likelihood.

**Comment by miloyip** [comment]
Fixed in ab250d21bc41b3a145d82330c92d30632980fb2f

**Comment by kkaefer** [comment]
\o/
